utils/utils.go

```diff
@@ -301,13 +301,75 @@ func CurrencyBalance(
 
 	liveAmount, err := types.ExtractAmount(liveBalances, currency)
 	if err != nil {
-		return nil, nil, fmt.Errorf(
+		return nil, nil, nil, fmt.Errorf(
 			"%w: could not get %s currency balance for %s",
 			err,
 			types.PrettyPrintStruct(currency),
 			types.PrettyPrintStruct(account),
 		)
 	}
 
-	return liveAmount, liveBlock, nil
+	return liveAmount, liveBlock, liveCoins, nil
+}
+
+// AccountBalanceRequest defines the required information
+// to get an account's balance.
+type AccountBalanceRequest struct {
+	Account  *types.AccountIdentifier
+	Network  *types.NetworkIdentifier
+	Currency *types.Currency
+}
+
+// AccountBalance defines an account's balance,
+// including either balance or coins, as well as
+// the block which this balance was fetched at.
+type AccountBalance struct {
+	Account       *types.AccountIdentifier
+	BalanceAmount *types.Amount
+	Coins         []*types.Coin
+	Block         *types.BlockIdentifier
+}
+
+// GetAccountBalances returns an array of AccountBalances
+// for an array of AccountBalanceRequests
+func GetAccountBalances(
+	ctx context.Context,
+	fetcher *fetcher.Fetcher,
+	balanceRequests []*AccountBalanceRequest,
+) ([]*AccountBalance, error) {
+	var accountBalances []*AccountBalance
+	for _, balanceRequest := range balanceRequests {
+		amount, block, coins, err := CurrencyBalance(
+			ctx,
+			balanceRequest.Network,
+			fetcher,
+			balanceRequest.Account,
+			balanceRequest.Currency,
+			nil,
+		)
+
+		if err != nil {
+			return nil, err
+		}
+
+		var accountBalance *AccountBalance
+		// Coins exist
+		if len(coins) > 0 {
+			accountBalance = &AccountBalance{
+				Account: balanceRequest.Account,
+				Coins:   coins,
+				Block:   block,
+			}
+		} else {
+			accountBalance = &AccountBalance{
+				Account:       balanceRequest.Account,
+				BalanceAmount: amount,
```

Comment:
You need to set this for UTXO chains too

Reply:
Balance tracking is always enabled (regardless of if Account-based or UTXO-based)